Modules/IO/TransformMINC/test/itkMINCTransformAdapterTest.cxx

```diff
@@ -230,6 +231,73 @@ itkMINCTransformAdapterTest(int argc, char * argv[])
   {
     itksys::SystemTools::ChangeDirectory(argv[1]);
   }
+
+
+  constexpr unsigned int InputDimension = 3;
+  constexpr unsigned int OutputDimension = 3;
+
+  using ParametersValueType = double;
+
+  using TransformAdapterType = itk::MINCTransformAdapter<ParametersValueType, InputDimension, OutputDimension>;
+
+  TransformAdapterType::Pointer xfm = TransformAdapterType::New();
+
+  ITK_EXERCISE_BASIC_OBJECT_METHODS(xfm, MINCTransformAdapter, Transform);
+
+  // Test exceptions
+  TransformAdapterType::InputVectorType vector{};
+  TransformAdapterType::InputPointType  point{};
+  ITK_TRY_EXPECT_EXCEPTION(xfm->TransformVector(vector, point));
+
+  TransformAdapterType::InputVnlVectorType vnlVector{};
+  ITK_TRY_EXPECT_EXCEPTION(xfm->TransformVector(vnlVector, point));
+
+  TransformAdapterType::InputVectorPixelType vectorPixel{};
+  ITK_TRY_EXPECT_EXCEPTION(xfm->TransformVector(vectorPixel, point));
+
+  TransformAdapterType::InputCovariantVectorType covariantVector{};
+  ITK_TRY_EXPECT_EXCEPTION(xfm->TransformCovariantVector(covariantVector, point));
+
+  ITK_TRY_EXPECT_EXCEPTION(xfm->TransformCovariantVector(vectorPixel, point));
+
+  TransformAdapterType::FixedParametersType fixedParams{};
+  ITK_TRY_EXPECT_EXCEPTION(xfm->SetFixedParameters(fixedParams));
+
+  TransformAdapterType::JacobianType jacobian{};
+  ITK_TRY_EXPECT_EXCEPTION(xfm->ComputeJacobianWithRespectToParameters(point, jacobian));
+
+  ITK_TRY_EXPECT_EXCEPTION(xfm->GetNumberOfParameters());
+
+  TransformAdapterType::ParametersType params{};
+  ITK_TRY_EXPECT_EXCEPTION(xfm->SetParameters(params));
+
+  ITK_TRY_EXPECT_EXCEPTION(xfm->GetParameters());
+
+  constexpr unsigned int InputDimensionExcp1 = 1;
+  constexpr unsigned int OutputDimensionExcp1 = 1;
+
+  using TransformAdapterTypeExcp11 =
+    itk::MINCTransformAdapter<ParametersValueType, InputDimensionExcp1, OutputDimensionExcp1>;
+
+  ITK_TRY_EXPECT_EXCEPTION(TransformAdapterTypeExcp11::New());
+
+  constexpr unsigned int InputDimensionExcp2 = 2;
+  constexpr unsigned int OutputDimensionExcp2 = 2;
+
+  using TransformAdapterTypeExcp22 =
+    itk::MINCTransformAdapter<ParametersValueType, InputDimensionExcp2, OutputDimensionExcp2>;
+
+  ITK_TRY_EXPECT_EXCEPTION(TransformAdapterTypeExcp22::New());
+
+  constexpr unsigned int InputDimensionExcp4 = 4;
+  constexpr unsigned int OutputDimensionExcp4 = 4;
+
+  using TransformAdapterTypeExcp44 =
+    itk::MINCTransformAdapter<ParametersValueType, InputDimensionExcp4, OutputDimensionExcp4>;
+
+  ITK_TRY_EXPECT_EXCEPTION(TransformAdapterTypeExcp44::New());
```

Comment:
The above three cases should ideally be done with a Cartesian product, but the template requires compile time constants. Maybe future versions of C++ will allow doing a `constexpr auto prod = CartesianProduct()` (?)